### PR TITLE
Tag navigation

### LINKED ISSUE #79: Tag navigation
Currently there is only `gi` that does the wrong thing.
We probably want
- `gi`: edit from front of first tag
- `ga`: edit from back of last tag

It would also be nice to use `:` to split a tag.

### PR DESCRIPTION
**Pull request by jeffhappily**
An attempt to fix https://github.com/NorfairKing/smos/issues/79

Please give me suggestions on my implementation!

### CODE REVIEW

File: smos-cursor/src/Smos/Cursor/Tag.hs [hunk 1]
```diff
@@ -80,3 +81,8 @@ tagCursorSelectPrevChar = tagCursorTextCursorL textCursorSelectPrev
 
 tagCursorSelectNextChar :: TagCursor -> Maybe TagCursor
 tagCursorSelectNextChar = tagCursorTextCursorL textCursorSelectNext
+
+tagCursorSplit :: TagCursor -> (TagCursor, TagCursor)
+tagCursorSplit = tagCursorTextCursorL $ \tc ->
+  let (a, b) = textCursorSplit tc
+   in (TagCursor a, b)
```
Comment (NorfairKing) [inline smos-cursor/src/Smos/Cursor/Tag.hs]: I would _really_ prefer if we don't use the `Functor` instance of tuples, but this is fine, I guess.
We may want to just add a comment about that.

File: smos-cursor/src/Smos/Cursor/Tags.hs [hunk 2]
```diff
@@ -186,3 +191,23 @@ tagsCursorSelectOrCreatePrevTag tc =
...
+
+tagsCursorSelectEndInSelectedTag :: TagsCursor -> TagsCursor
+tagsCursorSelectEndInSelectedTag = tagsCursorSelectedTagL %~ tagCursorSelectEnd
+
+tagsCursorSplit :: TagsCursor -> TagsCursor
+tagsCursorSplit tc =
+  let tagc = tc ^. tagsCursorSelectedTagL
+      (first, second) = tagCursorSplit tagc
+   in tc
+        & tagsCursorSelectedTagL %~ const first
```
Comment (NorfairKing) [inline smos-cursor/src/Smos/Cursor/Tags.hs]: `%~ const` is `.~` I think.

File: smos/src/Smos/Actions/Entry.hs [hunk 3]
```diff
@@ -100,10 +101,20 @@ entrySelectTags =
     { actionName = "entrySelectTags",
```
Comment (NorfairKing) [inline smos/src/Smos/Actions/Entry.hs]: Let's rename this to `entrySelectTagsFromStart` to be nicely symmetrical with `entrySelectTagsFromBack`.
",

**Comment by NorfairKing** [comment]
@jeffhappily It looks like this works exactly as intended. Great job. Looking forward to merge this.
I have a few minor comments but I can already merge this if you prefer. Just let me know.

File: smos-cursor/src/Smos/Cursor/Tag.hs (hunk 1 above)
Comment (jeffhappily) [reply smos-cursor/src/Smos/Cursor/Tag.hs]: Is there any better way to do it instead of using tuples?

File: smos/src/Smos/Actions/Entry.hs (hunk 3 above)
Comment (jeffhappily) [reply smos/src/Smos/Actions/Entry.hs]: I was thinking about this as well, but I wasn't so sure that is it a good idea to rename existing action

File: smos-cursor/src/Smos/Cursor/Tag.hs (hunk 1 above)
Comment (NorfairKing) [reply smos-cursor/src/Smos/Cursor/Tag.hs]: You can do explicit wrapping and unwrapping. They're just newtypes anyway.

Comment (jeffhappily) [reply smos-cursor/src/Smos/Cursor/Tag.hs]: Do you mean something like this?
```haskell
tagCursorSplit (TagCursor tc) =
  let (a, b) = textCursorSplit tc
   in (TagCursor a, TagCursor b)
```

Comment (NorfairKing) [reply smos-cursor/src/Smos/Cursor/Tag.hs]: Yes that's right.

**Comment by jeffhappily** [comment]
@NorfairKing I've fixed all the comments

**Review by NorfairKing — APPROVED** [review]
LGTM, merging.

**Pull request merged**